docs/DesignDocs/SGX_QuoteVerify_Integration.md

```diff
@@ -0,0 +1,488 @@
+`format_id` and `opt_params` for forward compatibility.
+
+```C
+oe_result_t oe_verify_quote_ocall(
+    [in] const oe_uuid_t* format_id,
+    [in, size=opt_params_size] const void* opt_params,
+    size_t opt_params_size,
+    [in, size=quote_size] const void* p_quote,
+    size_t quote_size,
+    oe_datetime_t expiration_check_date,
+    [out] uint32_t *p_collateral_expiration_status,
+    [out] uint32_t *p_quote_verification_result,
+    [in, out, size=qve_report_size] void* p_qve_report_info,
+    size_t qve_report_size,
+    [out, size=supplemental_data_size] void* p_supplemental_data,
+    size_t supplemental_data_size,
+    [out] size_t* p_supplemental_data_size_out);
+```
+
+### Update implementation of existing plugin functions
+
+These plugin functions include `_verify_evidence()`, `_verify_report()`
+and `oe_verify_quote_with_sgx_endorsements()`.
+
+- Update API `oe_verify_quote_with_sgx_endorsements()` implementation as below:
+  - Input parameter doesn't change.
+  - In host-side plugin:
+    - Set input parameter `p_qve_report_info` to NULL, then call host-side
+      function `oe_sgx_verify_quote()`, which will call SGX QVL library to
+      verify quote.
+  - In enclave-side plugin:
+    - Construct structure `sgx_ql_qe_report_info_t` as below, `nonce` and
+      `app_enclave_target_info` are input, qve_report is QvE report which
+      target to application enclave.
+    ```C
+    typedef struct _sgx_ql_qe_report_info_t {
+      sgx_quote_nonce_t nonce;
+      sgx_target_info_t app_enclave_target_info;
+      sgx_report_t qve_report;
+    }sgx_ql_qe_report_info_t;
+    ```
+    - Call OCALL function `oe_verify_quote_ocall()` to host-side plugin library
+      to verify SGX quote.
+    - With all the QvE return values, call new API.
+      `oe_verify_qve_report_and_identity()` (described below) to
+      verify QvE report and identity.
+- In API `_verify_evidence()`, remove function call `oe_get_sgx_endorsements()`,
+  and only call `oe_parse_sgx_endorsements()` if caller provides an endorsement buffer
+- In API `_verify_report()` and relevant internal APIs, such as
+  `oe_verify_report_internal()` and `oe_verify_sgx_quote()`, remove the code that
+  loads QPL and retrieves endorsements.
+
+### Add function in enclave-side plugin for QvE Identity verification
+
+For enclave-side verifier plugin, after quote verification, it needs to verify that the result
+comes from a trusted QvE.
+
+Intel® SGX SDK provides a library named `sgx_dcap_tvl` to help the verifier to
+verify QvE's identity, you can refer to [source file](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteVerification/dcap_tvl/sgx_dcap_tvl.cpp)
+
+This library uses hardcoded QvE identity values, because:
+- There is no need for x509 and JSON parser in verifier enclave's TCB.
+- Most of QvE's identity attributes only change infrequently.
+
+The only identity info may change more often is QvE's ISV SVN, so the verifier needs
```

Comment:
Isn't it QVL's responsibility for checking current QvE's product id, ISV SVN, etc against QvE Identity data from [QvE Identity at Intel® Provisioning
Cert Server](https://api.portal.trustedservices.intel.com/documentation#pcs-qve-identity-v2)? I am afraid hardcoded QvE identity values do not scale.

Reply:
Because QVL is an untrusted component, user cannot rely on QVL to verify QvE's identity. The good practice is to verify QvE's identity in user's enclave.

In the other word, an attacker can use faked QVL & QvE, if QvE's identity is checked by QVL, then user would have no way to find it's a faked QvE.

Reply:
Just to clarify, this is just a description of the Intel implementation of the enclave side trusted library for using the QVE, and we're just porting it into OE, correct? (i.e. we're not changing the internal logic of `sgx_verify_qve_report_and_identity()` when we import it as `oe_verify_qve_report_and_identity()`)

Reply:
@CodeMonkeyLeet , yes, the proposal is about porting the trusted library into OE, and the internal logic would not changed.